File: jobapplication/angularapp/src/app/services/application.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Application } from '../models/application.model';
import { apiUrl } from 'src/apiconfig';

@Injectable({
  providedIn: 'root',
})
export class ApplicationService {
  private baseUrl = `${apiUrl}/applications`; // Backend API URL

  constructor(private http: HttpClient) {}

  getAllApplications(): Observable<Application[]> {
    return this.http.get<Application[]>(this.baseUrl);
  }

  getApplicationById(id: number): Observable<Application> {
    return this.http.get<Application>(`${this.baseUrl}/${id}`);
  }

  saveApplication(application: any): Observable<Application> {
    console.log(application);
    
    return this.http.post<Application>(this.baseUrl, application);
  }

  getApplicationsByUserId(userId: number): Observable<Application[]> {
    return this.http.get<Application[]>(`${this.baseUrl}/user/${userId}`);
  }

  updateApplication(
    id: number,
    application: Application
  ): Observable<Application> {
    return this.http.put<Application>(`${this.baseUrl}/${id}`, application);
  }

  deleteApplication(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: jobapplication/springapp/src/main/java/com/examly/springapp/service/ApplicationService.java
package com.examly.springapp.service;

import com.examly.springapp.exception.AlreadyAppliedForJobException;
import com.examly.springapp.model.Application;
import java.util.List;
 
public interface ApplicationService {
    List<Application> getAllApplications();
    Application getApplicationById(int id);
    Application saveApplication(Application application) throws AlreadyAppliedForJobException;
    Application updateApplication(int id, Application application);
    void deleteApplication(int id);
    public List<Application> getApplicationsByUserId(int userId);
}  